src/run/storage/storage.cts

```diff
@@ -0,0 +1,68 @@
+// This is storage module that rest of modules should interact with.
+// Remaining modules in storage directory are implementation details
+// and should not be used directly outside of this directory.
+// There is eslint `no-restricted-imports` rule to enforce this.
+
+import { type BlobType } from '../../shared/blob-types.cjs'
+import { getTracer } from '../handlers/tracer.cjs'
+
+import { getRegionalBlobStore } from './regional-blob-store.cjs'
+import { getRequestScopedInMemoryCache } from './request-scoped-in-memory-cache.cjs'
+
+const encodeBlobKey = async (key: string) => {
+  const { encodeBlobKey: encodeBlobKeyImpl } = await import('../../shared/blobkey.js')
+  return await encodeBlobKeyImpl(key)
+}
+
+export const getMemoizedKeyValueStoreBackedByRegionalBlobStore = (
```

Comment:
I'm hopeful that the 3 implementations of this don't diverge too much. If we find ourselves adding any if statements in here for each of the types, the responsible thing to do quickly becomes splitting off.

Reply:
If we can move size calculation out of `storage` directory, we could add similar eslint rule like I did to not allow imports other than `BlobType` from the module that exports it in that directory to enforce storage not caring about details of individual types and only enforce that we use allowed union of types